src/index.ts

```diff
@@ -530,7 +530,24 @@ export function useState<S>(
             RootPath,
             () => setValue({ state: value.state }),
             value.state);
-        React.useEffect(() => () => value.state.destroy(), []);
+        const renders = React.useRef(0);
```

Comment:
Nice clever fix. Yes, it will impact the performance, unfortunately. However, there is a way around it. Define module-global isDevelopment boolean flag and execute the proposed code only when it is true, otherwise keep the original one under release/production mode. And please let me know if tests are passing with it (I understand they should be). And I will merge after (if you complete it within next 12 hours, I will merge it today, otherwise only after New Year holidays :) )